tests/integration/test_charm.py

```diff
@@ -95,15 +102,11 @@ async def test_application_primary(ops_test: OpsTest):
 
     assert number_of_primaries == 1, "more than one primary in replica set"
 
-    leader_id = await get_leader_id(ops_test)
-    assert (
-        primary == f"mongodb-k8s-{leader_id}.mongodb-k8s-endpoints:27017"
-    ), "primary not leader on deployment"
-
```

Comment:
Some context for @Mehdi-Bendriss  

@dmitry-ratushnyy and I discussed this. So this check was designed to test if the juju leader is the one that initialises the replica set (whoever initialises the replica set is the primary). But if the user provides their own cluster then this isn't a guarantee.

I think this case can be removed entirely, since I don't know how crucial it is to check that "the juju leader is the one that initialises the replica set " - If it is crucial we can add it back in and add a check that basically says:

```
if not user_deployed_cluster:
     leader_id = await get_leader_id(ops_test)
        assert (
            primary == f"mongodb-k8s-{leader_id}.mongodb-k8s-endpoints:27017"
        ), "primary not leader on deployment"
```

Reply:
@Mehdi-Bendriss does 👍 mean keep as is or add this check?